src/types/utils.cpp

```diff
@@ -15,12 +15,14 @@ using namespace Microsoft::Console;
 // - a string representation of the GUID. On failure, throws E_INVALIDARG.
 std::wstring Utils::GuidToString(const GUID guid)
 {
+    // 39: 38, plus swprintf needs somewhere to store the NUL terminator.
     std::array<wchar_t, 39> guid_cstr;
     const int written = swprintf(guid_cstr.data(), guid_cstr.size(), L"{%08x-%04x-%04x-%02x%02x-%02x%02x%02x%02x%02x%02x}", guid.Data1, guid.Data2, guid.Data3, guid.Data4[0], guid.Data4[1], guid.Data4[2], guid.Data4[3], guid.Data4[4], guid.Data4[5], guid.Data4[6], guid.Data4[7]);
 
     THROW_HR_IF(E_INVALIDARG, written == -1);
 
-    return std::wstring(guid_cstr.data(), guid_cstr.size());
+    // size - 1: The returned string should not be created including the NUL terminator.
+    return std::wstring(guid_cstr.data(), guid_cstr.size() - 1);

```

Comment:
Perhaps the following is also a good choice, you don't need to use std::array. If there is std::string::__resize_default_init like libc++, you can save a memory copy in this code.

```c++
constexpr size_t uuidlen=38;
std::wstring guid_cstr;
guid_cstr.resize(uuidlen+2);
const int written = swprintf(guid_cstr.data(), guid_cstr.size(), L"{%08x-%04x-%04x-%02x%02x-%02x%02x%02x%02x%02x%02x}", guid.Data1, guid.Data2, guid.Data3, guid.Data4[0], guid.Data4[1], guid.Data4[2], guid.Data4[3], guid.Data4[4], guid.Data4[5], guid.Data4[6], guid.Data4[7]);
THROW_HR_IF(E_INVALIDARG, written == -1);
guid_str.resize(uuidlen);
return guid_str;

```

https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_cat.cc

Reply:
You're not wrong that this could save a copy, but this is also not a terribly hot code path. /shrug